src/sentry/static/sentry/app/components/substances/workflowFilter.jsx

```diff
@@ -0,0 +1,153 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import {debounce} from 'lodash';
+
+import SelectControl from 'app/components/forms/selectControl';
+
+class WorkflowFilter extends React.Component {
+  static propTypes = {
+    tag: PropTypes.object.isRequired,
+    value: PropTypes.string,
+    onSelect: PropTypes.func,
+  };
+
+  static tagValueToSelectFormat = ({value}) => {
+    return {
+      value,
+      label: value,
+    };
+  };
+
+  static defaultProps = {
+    tag: {},
+    value: '',
+  };
+
+  constructor(...args) {
+    super(...args);
+    this.state = {
+      query: '',
+      isLoading: false,
+      value: this.props.value,
+      textValue: this.props.value,
+    };
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.value !== this.state.value) {
+      this.setState({
+        value: nextProps.value,
+        textValue: nextProps.value,
+      });
+    }
+  }
+
+  handleLoadOptions = () => {};
+
+  handleChangeInput = e => {
+    const value = e.target.value;
+    this.setState({
+      textValue: value,
+    });
+    this.debouncedTextChange(value);
+  };
+
+  debouncedTextChange = debounce(function(text) {
+    this.handleChange(text);
+  }, 150);
+
+  handleOpenMenu = () => {
+    if (this.props.tag.predefined) {
+      return;
+    }
+
+    this.setState(
+      {
+        isLoading: true,
+      },
+      this.handleLoadOptions
+    );
+  };
+
+  handleChangeSelect = valueObj => {
+    const value = valueObj ? valueObj.value : null;
+    this.handleChange(value);
+  };
+
+  handleChangeSelectInput = value => {
+    this.setState(
+      {
+        textValue: value,
+      },
+      this.handleLoadOptions
+    );
+  };
+
+  handleChange = value => {
+    const {onSelect, tag} = this.props;
+
+    this.setState(
+      {
+        value,
+      },
+      () => {
+        onSelect(tag, value);
+      }
+    );
+  };
+  // <SelectControl
+  //   filterOptions={(options, filter, currentValues) => options}
+  //   placeholder="Select workflow"
+  //   value={this.state.value}
+  //   onChange={this.handleChangeSelect}
+  //   isLoading={this.state.isLoading}
+  //   onInputChange={this.handleChangeSelectInput}
+  //   onOpen={this.handleOpenMenu}
+  //   autoload={false}
+  //   noResultsText={this.state.isLoading ? t('Loading...') : t('No results found')}
+  //   options={
+  //     tag.predefined
+  //       ? tag.values &&
+  //         tag.values.map(value => ({
+  //           value,
+  //           label: value,
+  //         }))
+  //       : this.state.options
+  //   }
+  // />
```

Comment:
Remove unused SelectControl?

Reply:
This file was removed in a later commit.